src/usr.bin/uuencode/uuencode.1
.\"	$ssdlinux: uuencode.1,v 1.1.1.1 2002/05/02 13:37:12 kanoh Exp $
.\"	$NetBSD: uuencode.1,v 1.9 1999/03/22 18:16:45 garbled Exp $
.\"
.\" Copyright (c) 1980, 1990, 1993
.\"	The Regents of the University of California.  All rights reserved.
.\"
.\" Redistribution and use in source and binary forms, with or without
.\" modification, are permitted provided that the following conditions
.\" are met:
.\" 1. Redistributions of source code must retain the above copyright
.\"    notice, this list of conditions and the following disclaimer.
.\" 2. Redistributions in binary form must reproduce the above copyright
.\"    notice, this list of conditions and the following disclaimer in the
.\"    documentation and/or other materials provided with the distribution.
.\" 3. All advertising materials mentioning features or use of this software
.\"    must display the following acknowledgement:
.\"	This product includes software developed by the University of
.\"	California, Berkeley and its contributors.
.\" 4. Neither the name of the University nor the names of its contributors
.\"    may be used to endorse or promote products derived from this software
.\"    without specific prior written permission.
.\"
.\" THIS SOFTWARE IS PROVIDED BY THE REGENTS AND CONTRIBUTORS ``AS IS'' AND
.\" ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
.\" IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
.\" ARE DISCLAIMED.  IN NO EVENT SHALL THE REGENTS OR CONTRIBUTORS BE LIABLE
.\" FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
.\" DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
.\" OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
.\" HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
.\" LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
.\" OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
.\" SUCH DAMAGE.
.\"
.\"     @(#)uuencode.1	8.1 (Berkeley) 6/6/93
.\"
.Dd March 19, 1999
.Dt UUENCODE 1
.Os
.Sh NAME
.Nm uuencode ,
.Nm uudecode
.Nd encode/decode a binary file
.Sh SYNOPSIS
.Nm
.Op Ar file
.Ar name
.Nm uudecode
.Op Fl p
.Op Ar file ...
.Sh DESCRIPTION
.Nm
and
.Nm uudecode
are used to transmit binary files over transmission mediums
that do not support other than simple
.Tn ASCII
data.
.Pp
.Nm
reads
.Ar file
(or by default the standard input) and writes an encoded version
to the standard output.
The encoding uses only printing
.Tn ASCII
characters and includes the
mode of the file and the operand
.Ar name
for use by
.Nm uudecode .
.Pp
.Nm uudecode
transforms
.Em uuencoded
files (or by default, the standard input) into the original form.
The resulting file is named
.Ar name
and will have the mode of the original file except that setuid
and execute bits are not retained; if the
.Fl p
option is specified, the data will be written to the standard output.
.Nm uudecode
ignores any leading and trailing lines.
.Sh EXAMPLES
The following example packages up a source tree, compresses it,
uuencodes it and mails it to a user on another system.
When
.Nm uudecode
is run on the target system, the file ``src_tree.tar.Z'' will be
created which may then be uncompressed and extracted into the original
tree.
.Pp
.Bd -literal -offset indent -compact
tar cf \- src_tree \&| compress \&|
uuencode src_tree.tar.Z \&| mail sys1!sys2!user
.Ed
.Sh DIAGNOSTICS
The 
.Nm uudecode
and 
.Nm
utilities exits 0 on success, and >0 if an error occurs.
.Sh SEE ALSO
.Xr compress 1 ,
.Xr mail 1 ,
.Xr uucp 1 ,
.Xr uuencode 5
.Sh STANDARDS
The 
.Nm uudecode
and
.Nm
utilities conform to 
.St -p1003.2-92 .
.Sh HISTORY
The
.Nm uudecode
and
.Nm
utilities appeared in
.Bx 4.0 .
.Sh BUGS
The encoded form of the file is expanded by 35% (3 bytes become 4 plus
control information).


src/ChangeLog
$ssdlinux: ChangeLog,v 1.492 2005/03/20 09:18:33 kanoh Exp $

Sun Mar 20 18:15:57 JST 2005  Tamotsu Kanoh <kanoh@kanoh.org>
	* Update GCC to version 3.3.5.

Fri Mar 18 11:27:00 JST 2005  Atsushi Yamagata <yamagata@plathome.co.jp>
	* Update findutils to version 4.2.20.

Fri Mar 18 09:57:30 JST 2005  Atsushi Yamagata <yamagata@plathome.co.jp>
	* 0.3-20050318 Release.

Wed Mar 16 17:00:49 JST 2005  Atsushi Yamagata <yamagata@plathome.co.jp>
	* Update strace to version 4.5.10.

Wed Mar 16 15:48:34 JST 2005  Atsushi Yamagata <yamagata@plathome.co.jp>
	* Update postfix to version 2.2.1.

Tue Mar 15 11:32:34 JST 2005  Atsushi Yamagata <yamagata@plathome.co.jp>
	* Update gettext to version 0.14.3.

Tue Mar 15 10:18:52 JST 2005  Atsushi Yamagata <yamagata@plathome.co.jp>
	* Update psmisc to version 21.6.

Mon Mar 14 12:48:35 JST 2005  Atsushi Yamagata <yamagata@plathome.co.jp>
	* Updated bind to version 9.2.5.

Mon Mar 14 10:15:00 JST 2005  Atsushi Yamagata <yamagata@plathome.co.jp>
	* Update ncurses to version 5.4-20050312.

Fri Mar 11 16:43:48 JST 2005  Atsushi Yamagata <yamagata@plathome.co.jp>
	* Update tnftp to version 20050103.

Fri Mar 11 09:58:54 JST 2005  Atsushi Yamagata <yamagata@plathome.co.jp>
	* 0.3-20050311 Release.

Thu Mar 10 17:44:49 JST 2005  Atsushi Yamagata <yamagata@plathome.co.jp>
	* Update postfix to version 2.2.0.

Thu Mar 10 16:11:46 JST 2005  Atsushi Yamagata <yamagata@plathome.co.jp>
	* Update iptables to version 1.3.1.

Thu Mar 10 14:14:08 JST 2005  Atsushi Yamagata <yamagata@plathome.co.jp>
	* Update openssh to version 4.0p1.

Mon Mar  7 10:15:38 JST 2005  Atsushi Yamagata <yamagata@plathome.co.jp>
	* Update ncurses to version 5.4-20050305.

Fri Mar  4 18:12:35 JST 2005  Atsushi Yamagata <yamagata@plathome.co.jp>
	* 0.3-20050304 Release.

Wed Mar  2 15:04:15 JST 2005  Atsushi Yamagata <yamagata@plathome.co.jp>
	* Update e2fsprogs to version 1.36.

Wed Mar  2 10:45:01 JST 2005  Atsushi Yamagata <yamagata@plathome.co.jp>
	* Update dhcp to version 3.0.2.

Mon Feb 28 10:14:46 JST 2005  Atsushi Yamagata <yamagata@plathome.co.jp>
	* Update ncurses to version 5.4-20050226.

Fri Feb 25 16:23:38 JST 2005  Atsushi Yamagata <yamagata@plathome.co.jp>
	* Update ex to version 050225.

Fri Feb 25 16:05:29 JST 2005  Atsushi Yamagata <yamagata@plathome.co.jp>
	* Update gettext to version 0.14.2.

Fri Feb 25 09:46:36 JST 2005  Atsushi Yamagata <yamagata@plathome.co.jp>
	* 0.3-20050225 Release.

Mon Feb 21 11:58:44 JST 2005  Atsushi Yamagata <yamagata@plathome.co.jp>
	* Update zlib to version 1.2.2.

Mon Feb 21 11:28:09 JST 2005  Atsushi Yamagata <yamagata@plathome.co.jp>
	* Update findutils to version 4.2.18.

Mon Feb 21 10:57:22 JST 2005  Atsushi Yamagata <yamagata@plathome.co.jp>
	* Update ncurses to version 5.4-20050219.

Fri Feb 18 10:39:24 JST 2005  Atsushi Yamagata <yamagata@plathome.co.jp>
	* 0.3-20050218 Release.

Tue Feb 15 18:33:14 JST 2005  Atsushi Yamagata <yamagata@plathome.co.jp>
	* 0.3-20050215 Release.

Mon Feb 14 12:03:37 JST 2005  Atsushi Yamagata <yamagata@plathome.co.jp>
	* Update iptables to version 1.3.0.

Mon Feb 14 11:03:45 JST 2005  Atsushi Yamagata <yamagata@plathome.co.jp>
	* Update file to version 4.13.

Mon Feb 14 10:40:03 JST 2005  Atsushi Yamagata <yamagata@plathome.co.jp>
	* Update libtool to version 1.5.14.

Mon Feb 14 10:33:00 JST 2005  Atsushi Yamagata <yamagata@plathome.co.jp>
	* Update automake to version 1.9.5.

Mon Feb 14 10:28:42 JST 2005  Atsushi Yamagata <yamagata@plathome.co.jp>
	* Update ncurses to version 5.4-20050212.

Thu Feb 10 18:02:01 JST 2005  Jun Todoroki <todoroki@plathome.co.jp>
	* Update usbustils to version 0.70 & add libusb-0.1.9.

Wed Feb  9 11:34:42 JST 2005  Atsushi Yamagata <yamagata@plathome.co.jp>
	* Update grub to version 0.96.

Wed Feb  9 14:36:01 JST 2005  Jun Todoroki <todoroki@plathome.co.jp>
	* Update madwifi to version 20050209.

Mon Feb  7 14:20:34 JST 2005  Atsushi Yamagata <yamagata@plathome.co.jp>
	* Update strace to version 4.5.9.

Mon Feb  7 13:47:22 JST 2005  Atsushi Yamagata <yamagata@plathome.co.jp>
	* Update sudo to version 1.6.8p7.

Mon Feb  7 11:17:34 JST 2005  Atsushi Yamagata <yamagata@plathome.co.jp>
	* Update cvs to version 1.11.19.

Mon Feb  7 10:47:05 JST 2005  Atsushi Yamagata <yamagata@plathome.co.jp>
	* Update libtool to version 1.5.12.

Mon Feb  7 10:38:15 JST 2005  Atsushi Yamagata <yamagata@plathome.co.jp>
	* Update ncurses to version 5.4-20050205.

Mon Jan 31 12:29:41 JST 2005  Atsushi Yamagata <yamagata@plathome.co.jp>
	* Update sed to version 4.1.4.

Mon Jan 31 12:19:56 JST 2005  Atsushi Yamagata <yamagata@plathome.co.jp>
	* Update ncurses to version 5.4-20050129.

Mon Jan 31 11:29:54 JST 2005  Atsushi Yamagata <yamagata@plathome.co.jp>
	* Update findutils to version 4.2.15.

Sat Jan 29 18:05:07 JST 2005  Atsushi Yamagata <yamagata@plathome.co.jp>
	* 0.3-20050129 Release.

Thu Jan 27 15:57:02 JST 2005  Atsushi Yamagata <yamagata@plathome.co.jp>
	* Update util-linux to version 2.12q.

Thu Jan 27 12:41:46 JST 2005  Atsushi Yamagata <yamagata@plathome.co.jp>
	* Update shadow to version 4.0.7.

Thu Jan 27 11:38:48 JST 2005  Atsushi Yamagata <yamagata@plathome.co.jp>
	* Update procps to version 3.2.5.
	
Thu Jan 27 11:18:56 JST 2005  Atsushi Yamagata <yamagata@plathome.co.jp>
	* Update ex to version 050119.

Thu Jan 27 11:01:33 JST 2005  Atsushi Yamagata <yamagata@plathome.co.jp>
	* Update dump to version 0.4b39.

Thu Jan 27 10:36:16 JST 2005  Atsushi Yamagata <yamagata@plathome.co.jp>
	* Update sed to version 4.1.3.

Wed Jan 26 17:08:50 JST 2005  Atsushi Yamagata <yamagata@plathome.co.jp>
	* Update ncurses to version 5.4-20050122.

Fri Jan 21 23:18:22 JST 2005  Atsushi Yamagata <yamagata@plathome.co.jp>
	* 0.3-20050121 Release.

Wed Jan 19 11:18:33 JST 2005  Atsushi Yamagata <yamagata@plathome.co.jp>
	* Update a patch for sendmail (Errata: 2005-01-17).

Mon Jan 17 11:04:35 JST 2005  Atsushi Yamagata <yamagata@plathome.co.jp>
        * Apply a patch for sendmail (Errata: 2005-01-15).


src/usr.bin/uuencode/uuencode.c
/*	$ssdlinux: uuencode.c,v 1.1.1.1 2002/05/02 13:37:12 kanoh Exp $	*/
/*	$NetBSD: uuencode.c,v 1.8 1997/10/20 02:51:01 lukem Exp $	*/

/*-
 * Copyright (c) 1983, 1993
 *	The Regents of the University of California.  All rights reserved.
 *
 * Redistribution and use in source and binary forms, with or without
 * modification, are permitted provided that the following conditions
 * are met:
 * 1. Redistributions of source code must retain the above copyright
 *    notice, this list of conditions and the following disclaimer.
 * 2. Redistributions in binary form must reproduce the above copyright
 *    notice, this list of conditions and the following disclaimer in the
 *    documentation and/or other materials provided with the distribution.
 * 3. All advertising materials mentioning features or use of this software
 *    must display the following acknowledgement:
 *	This product includes software developed by the University of
 *	California, Berkeley and its contributors.
 * 4. Neither the name of the University nor the names of its contributors
 *    may be used to endorse or promote products derived from this software
 *    without specific prior written permission.
 *
 * THIS SOFTWARE IS PROVIDED BY THE REGENTS AND CONTRIBUTORS ``AS IS'' AND
 * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
 * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
 * ARE DISCLAIMED.  IN NO EVENT SHALL THE REGENTS OR CONTRIBUTORS BE LIABLE
 * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
 * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
 * OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
 * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
 * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
 * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
 * SUCH DAMAGE.
 */

#include <sys/cdefs.h>
/*
#ifndef lint
__COPYRIGHT("@(#) Copyright (c) 1983, 1993\n\
	The Regents of the University of California.  All rights reserved.\n");
#endif */ /* not lint */

/*
#ifndef lint
#if 0
static char sccsid[] = "@(#)uuencode.c	8.2 (Berkeley) 4/2/94";
#else
__RCSID("$NetBSD: uuencode.c,v 1.8 1997/10/20 02:51:01 lukem Exp $");
#endif
#endif */ /* not lint */

/*
 * uuencode [input] output
 *
 * Encode a file so it can be mailed to a remote system.
 */
#include <sys/types.h>
#include <sys/stat.h>
#include <err.h>
#include <errno.h>
#include <locale.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>

int	main __P((int, char **));
static void encode __P((void));
static void usage __P((void));

int
main(argc, argv)
	int argc;
	char *argv[];
{
	struct stat sb;
	int mode;

	mode = 0;
	setlocale(LC_ALL, "");

	while (getopt(argc, argv, "") != -1)
		usage();
	argv += optind;
	argc -= optind;

	switch(argc) {
	case 2:			/* optional first argument is input file */
		if (!freopen(*argv, "r", stdin) || fstat(fileno(stdin), &sb))
			err(1, "%s", *argv);
#define	RWX	(S_IRWXU|S_IRWXG|S_IRWXO)
		mode = sb.st_mode & RWX;
		++argv;
		break;
	case 1:
#define	RW	(S_IRUSR|S_IWUSR|S_IRGRP|S_IWGRP|S_IROTH|S_IWOTH)
		mode = RW & ~umask(RW);
		break;
	case 0:
	default:
		usage();
	}

	(void)printf("begin %o %s\n", mode, *argv);
	encode();
	(void)printf("end\n");
	if (ferror(stdout))
		err(1, "write error");
	exit(0);
}

/* ENC is the basic 1 character encoding function to make a char printing */
#define	ENC(c) ((c) ? ((c) & 077) + ' ': '`')

/*
 * copy from in to out, encoding as you go along.
 */
static void
encode()
{
	int ch, n;
	char *p;
	char buf[80];

	while ((n = fread(buf, 1, 45, stdin)) > 0) {
		ch = ENC(n);
		if (putchar(ch) == EOF)
			break;
		for (p = buf; n > 0; n -= 3, p += 3) {
			ch = *p >> 2;
			ch = ENC(ch);
			if (putchar(ch) == EOF)
				break;
			ch = ((*p << 4) & 060) | ((p[1] >> 4) & 017);
			ch = ENC(ch);
			if (putchar(ch) == EOF)
				break;
			ch = ((p[1] << 2) & 074) | ((p[2] >> 6) & 03);
			ch = ENC(ch);
			if (putchar(ch) == EOF)
				break;
			ch = p[2] & 077;
			ch = ENC(ch);
			if (putchar(ch) == EOF)
				break;
		}
		if (putchar('\n') == EOF)
			break;
	}
	if (ferror(stdin))
		err(1, "read error.");
	ch = ENC('\0');
	(void)putchar(ch);
	(void)putchar('\n');
}

static void
usage()
{
	(void)fprintf(stderr,"usage: uuencode [infile] remotefile\n");
	exit(1);
}
